### PR TITLE
Enhancement of Authentication Handling and Guest Access

### PR DESCRIPTION
**Pull request by leagrieder**
This pull request introduces updates to the handling of user authentication and guest access across the application, focusing primarily on enhancing user experience and securing user-specific functionalities. The changes are aimed at allowing users to navigate the app as guests and prompting login when necessary, alongside improvements in profile management and navigation.

Key Changes:

1. Guest Access in Login Screen:
- Added a "Continue As Guest" button to the `LoginScreen`. This allows users to explore certain features of the app without logging in

2. Dynamic Authentication Checks:
- Integrated checks across the `HomeScreen` to verify user authentication status before allowing access to user-specific features like viewing or editing profiles, and organizing events/creating events.

3. Used `AccountViewModel` to manage user login status and to prompt login actions dynamically based on the user's actions.
ViewModel Adjustments:
- Enhanced `AccountViewModel` to include methods for checking user authentication (isUserLoggedIn) and for prompting users to log in (promptLogin), ensuring a cohesive user experience.
- Adjusted ViewModel to handle new authentication flows introduced by guest access.

4. In `HomeScreen`, I noticed that we were still using Dummy User data to show a a "by default" profile picture, I modified this screen so that it dynamically checks whether the current user (if not a guest user) has a profile picture and if he does it will show this profile picture on the HomeScreen 

These changes aim to make the application more accessible to first-time users while maintaining security and functionality for returning users.

After Gregory's review, the login screen now looks something like this: 
image

Next Steps:

- Conduct a thorough audit of the new authentication flows to ensure robust security measures are in place. This includes verifying that there are no loopholes allowing unauthorized access to user-specific features and functionalities.
- Expand Test Coverage

This PR is open for review, and I look forward to your suggestions and feedback to make further improvements.

### CODE REVIEW

**Review by Kuriv2001 — COMMENTED** [review]
Finally a person interested in events can look at them without creating an account, well done! This will help reduce the friction of a new user joining the app, who never used the app before (firs time user).

I like that we can now see the actual profile picture of a user on the HomeScreen.

The original functionality still seems to be working well and hasn't been affected by the new way to look at events. I also think the change of Toasts when joining events is justified and makes more sense like this, not to overload the user with Toasts and make the app look less responsive.

The new guest mode is well implemented and very robust. Trying the app I never ran into any problems and always got redirected in a safe way to the login screen, when I shouldn't be able to do an action.

In my eyes, the button on the Login Screen should be made to look similar to the "Sign in  with Google" Button (just the shape and size) and have its String as a language resource(like the Sign in Button).

In the AccountViewModel just a little change is needed to put Route.Login_Screen instead of "Login" on line 130.

File: app/src/main/java/com/monkeyteam/chimpagne/ui/HomeScreen.kt [hunk 1]
```diff
@@ -24,26 +25,31 @@ import androidx.compose.ui.unit.sp
 import com.monkeyteam.chimpagne.R
 import com.monkeyteam.chimpagne.ui.components.ChimpagneButton
 import com.monkeyteam.chimpagne.ui.components.ProfileIcon
-import com.monkeyteam.chimpagne.ui.components.User
 import com.monkeyteam.chimpagne.ui.navigation.NavigationActions
 import com.monkeyteam.chimpagne.ui.navigation.Route
+import com.monkeyteam.chimpagne.viewmodels.AccountViewModel
 
+@SuppressLint("StateFlowValueCalledInComposition")
```
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/HomeScreen.kt:32]: to suppress the warning:  val uiState by accountViewModel.uiState.collectAsState() (see MyEventsScreen for instance)

**Review by QuicheLorraine3600 — CHANGES_REQUESTED** [review]
🤩

File: app/src/main/java/com/monkeyteam/chimpagne/viewmodels/AccountViewModel.kt [hunk 2]
```diff
@@ -120,6 +125,17 @@ class AccountViewModel(database: Database) : ViewModel() {
     _uiState.value = _uiState.value.copy(tempProfilePicture = uri)
     Log.d("AccountViewModel", "Updated Profile Picture to $uri")
   }
+
+  // This function checks if the user is logged in or not and shows a Toast and redirects to the
+  // Login screen if the user is a guest
+  fun promptLogin(context: Context, navActions: NavigationActions) {
```
[the remark this answers is not in the document]
Comment (leagrieder) [reply app/src/main/java/com/monkeyteam/chimpagne/viewmodels/AccountViewModel.kt:131]: Thanks for the feedback!

**Comment by Athanaze** [comment]
As discussed irl, this is ready for merging 👍

**Review by QuicheLorraine3600 — APPROVED** [review]

**Pull request merged**